### ISSUE TITLE
Sites on review apps are not properly defined

### ISSUE THREAD

**Issue by data-sync-user** [MozillaFoundation/foundation.mozilla.org#12255]
Review apps are configured now to create Wagtail `site` instances based on staging:

!image-20240426-161823.png|width=1022,height=295,alt="image-20240426-161823.png"!

This makes it so that any page links will redirect the user to the “staging” instance instead of staying inside the review app. It makes the main nav menu not work properly.

This should be changed to have the review app’s URL as the main site:

!image-20240426-162004.png|width=973,height=325,alt="image-20240426-162004.png"!

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/TP1-569)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/62050">image-20240426-161823.png</a> | <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/62051">image-20240426-162004.png</a>

**Comment by data-sync-user** [comment]
➤ Daniel Miranda commented:

After looking into this, it appears that the error is only reproduced when recreating prod/staging on review apps via mofo-cron, so the code change is going to need to take place in that repo!

**Comment by data-sync-user** [comment]
➤ Adalberto Vazquez Roman commented:

Improving the Process of Recreating Production Database on Heroku Review AppsAs Daniel Miranda highlighted previously, this issue pertains to the process of recreating the Production database on Heroku Review Apps via the mofo-cron repository.

Issue OverviewThe root of the problem lies in using the same script for both syncing the Staging database with the Production database and syncing the Review Apps database with the Production database. Although these processes may seem very similar, this dual-purpose use requires some adjustments to the original script:

  1. URL Replacement in cleanup.sql:
The primary issue comes from the cleanup.sql script that is executed within tasks/clone_foundation_site/task.sh. This script consistently replaces domains with the Staging site's URL, causing all page links to redirect users to the Staging instance. To resolve this, we need to update the script to replace the URL with the Review App’s URL, ensuring correct redirection.
Please refer to the code here. ( https://github.com/MozillaFoundation/mofo-cron/blob/6323ce2b00c0c30b6ee23d30654c63f3088c9593/tasks/clone_foundation_site/cleanup.sql#L17-L32 )

  1. Optimizing S3 Bucket Sync:
As an enhancement, I recommend commenting out the code lines that sync the S3 Staging Bucket with the Production one. All Review Apps connect to the same S3 bucket as Staging. Given that the mofo-cron task runs every Monday, omitting this step can save time and resources. While there is a minor risk of some blog entries or pages lacking images, the critical data will remain intact, enabling effective testing without significant issues.
Please refer to the lines here. ( https://github.com/MozillaFoundation/mofo-cron/blob/6323ce2b00c0c30b6ee23d30654c63f3088c9593/tasks/clone_foundation_site/task.sh#L108-L109 )

  1. Creating a Separate Script:
To further improve the process, I strongly suggest creating a dedicated script file, possibly within the mofo-cron repository or the foundation.mozilla.org repo. This separation ensures we avoid accidental modifications to critical scripts. The tasks/clone_foundation_site/task.sh file's primary purpose is to sync Staging and Production environments. In contrast, our goal is to copy essential parts of the Production database to Review Apps for testing new features or bug fixes. This doesn't require an exact replica. A separate script could streamline the process by using an invoke command with the Heroku Review App name, automating the necessary adjustments without additional configuration.

These recommendations aim to make the process more maintainable and minimize the need for modifying existing files to fit our requirements. Investing time in creating a separate script aligns with our need for a "newer" site version for testing purposes.

In summary, my recommendations are:

  * Create a new file to execute a script that has the sole purpose of copying relevant Production database data to the Heroku Review apps. This relevant copy can be made by taking the tasks/clone_foundation_site/task.sh file as a baseline, properly replacing URLs depending on the Review App name inside cleanup.sql, and removing the step of syncing S3 buckets.

Please share your thoughts and feedback on these suggestions! Tyler Denton Daniel Miranda Rob DiVincenzo Ramon Ramos

**Comment by data-sync-user** [comment]
➤ Tyler Denton commented:

Thank you for this Adalberto Vazquez Roman , excellent run-down. I’m all for creating a separate script for RAs if it helps keep things safer and more readable through separation (even if it does cost us a bit of non-DRY code). I’m curious too if this will alleviate the “database disruption imminent” alert email that we get from Heroku concerning the RA Postgres row count being larger than the instance allows if there’s some opportunities to trim things down to resolve that? That’d be neat.

**Comment by data-sync-user** [comment]
➤ Adalberto Vazquez Roman commented:

After reviewing the mail message we received from the Review Apps, I propose we address the issue by adding steps to a new script. These steps will automatically delete rows from the backup we got from the Production DB that are not important to have inside the Review Apps.

For example, we can retain only the most recent blog entries (last 6 months) and apply the same rule to other pages. To implement this, I will need assistance from an experienced Wagtail developer. However, would be great to handle this work in a separate ticket to avoid increasing the scope of the current one.

Once this script is created and tested, I will open a new ticket to address the "database disruption imminent" alert mail immediately with the extra steps mentioned.

What do you think Tyler Denton ?

**Comment by data-sync-user** [comment]
➤ Tyler Denton commented:

For sure, thanks again Adal. Creating a new ticket and seeing if the juice is worth the squeeze sounds like the best approach
